src/Query/Builder.php

```diff
@@ -596,6 +599,11 @@ public function aggregate($function = null, $columns = ['*'])
         $this->columns            = $previousColumns;
         $this->bindings['select'] = $previousSelectBindings;
 
+        // When the aggregation is per group, we return the results as is.
+        if ($this->groups) {
+            return $results;
+        }
```

Comment:
Trying to support this feature in Laravel, so that 
`withCount` on hybrid relations could work: https://github.com/laravel/framework/pull/53209

Reply:
Can you just not add withCount on hybrid relations, its a blocker.  Also a suggestion It would be better for hybrid to be a separate package instead of being integrated into laravel-mongodb if that could ease the maintenance and focus for other mongodb eloquent methods.